plugins/glustershd/rest.go

```diff
@@ -0,0 +1,198 @@
+package glustershd
+
+import (
+	"net/http"
+
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/pkg/api"
+	"github.com/gluster/glusterd2/pkg/errors"
+
+	"github.com/gorilla/mux"
+	"github.com/pborman/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+func isVolReplicate(vType volume.VolType) bool {
+	if vType == volume.Replicate || vType == volume.Disperse || vType == volume.DistReplicate || vType == volume.DistDisperse {
+		return true
+	}
+
+	return false
+}
+
+func glustershEnableHandler(w http.ResponseWriter, r *http.Request) {
+	// Implement the help logic and send response back as below
+	p := mux.Vars(r)
+	volname := p["name"]
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	//validate volume name
+	v, err := volume.GetVolume(volname)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusNotFound, errors.ErrVolNotFound.Error(), api.ErrCodeDefault)
+		return
+	}
+	// Store initial volinfo before changing the HealFlag
+	oldvolinfo := v
```

Comment:
`v` is a pointer i.e of type `*volume.Volinfo`. So `oldinfo` will also b a pointer pointing to the same struct. So the following will actually change the underlying actual structure.
```go
v.HealEnabled = true
```

What you have to do, is this something like this:
```go
tmp := *v
oldvolinfo := &tmp
```

In the above example, `tmp` is a copy of struct pointed by `v`.